examples/provider/keyvalue-inmemory/README.md

```diff
@@ -1,9 +1,66 @@
-# Keyvalue inmemory golang provider
+# In-Memory Key-Value Storage Provider
 
-[This provider example](https://github.com/wasmCloud/go/tree/main/examples/provider/keyvalue-inmemory)
-is implements `wrpc:keyvalue/store@0.2.0-draft`.
+[This capability provider example](https://github.com/wasmCloud/go/tree/main/examples/provider/keyvalue-inmemory) implements `wrpc:keyvalue/store@0.2.0-draft` to provide key-value functionality (backed in-memory) for components that **import** the same interface. 
 
-## Notable files
+## 👟 Run the Go project
 
-- main.go is a simple binary that sets up an errGroup to handle running the provider's primary requirements: executing as a standaline binary based on data received on stdin, handling RPC and connecting to a wasmCloud lattice.
-- keyvalue.go implements the required functions to conform to `wasi:keyvalue/store`. If the functions as specified in the `./wit` directory are not implemented, this provider will fail to build.
+Clone the [wasmCloud/go repository](https://github.com/wasmcloud/go): 
+
+```shell
+git clone https://github.com/wasmCloud/go.git
+```
+
+Change directory to `examples/provider/keyvalue-inmemory`:
+
+```shell
+cd examples/provider/keyvalue-inmemory
+```
+
+In addition to the standard elements of a Go project, the example directory includes the following files and directories:
+
+- `/bindings`: Directory for Go bindings of [interfaces](https://wasmcloud.com/docs/concepts/interfaces)
+- `/wit`: Directory for WebAssembly Interface Type (WIT) packages that define interfaces
+- `wasmcloud.toml`: Configuration file for a wasmCloud application
+
+The example includes a shell script to simplify running the Go project:
+
+```shell
+./run.sh
+```
+
+You can also perform each step manually:
+
+```shell
+host_data='{"lattice_rpc_url": "0.0.0.0:4222", "lattice_rpc_prefix": "default", "provider_key": "keyvalue-inmemory", "link_name": "default","log_level": "debug"}' 
+echo $host_data | base64 | go run ./
+```
+
+## 🛠️ Build the provider
+
+The example includes a shell script to simplify the build process:
```

Comment:
`wash build` should be able to build the provider and I think it does all of these steps so I don't remember why we have the `build.sh` file. I just ran `wash build` in this directory and it built a provider